### PR TITLE
PARQUET-1494: [C++] Recognize statistics built with UNSIGNED sort order by parquet-mr 1.10.0 onwards

### PR DESCRIPTION
**Pull request by zilder**
Fixes the issue with min-max statistics built by parquet-mr 1.10+:
https://issues.apache.org/jira/browse/PARQUET-1494

### CODE REVIEW

File: cpp/src/parquet/metadata.cc
Comment (majetideepak) [inline cpp/src/parquet/metadata.cc]: Should be `parquet-mr`

**Review by majetideepak — CHANGES_REQUESTED** [review]
Edit: Can you add the file generated by parquet-mr to the data folder in the parquet-testing submodule https://github.com/apache/arrow/tree/master/cpp/submodules and add a test as well?

File: cpp/src/parquet/metadata.cc
Comment (zilder) [reply cpp/src/parquet/metadata.cc]: sure, sorry

**Comment by zilder** [comment]
Added a test case.

**Comment by zilder** [comment]
Btw, thanks to @sumermann for helping to investigate the issue and @alkagin for providing the sample parquet file for the test case.

**Comment by majetideepak** [comment]
@zilder looks like you need to run clang-format
```
/home/travis/build/apache/arrow/cpp/src/parquet/statistics-test.cc had clang-format style issues
--- /home/travis/build/apache/arrow/cpp/src/parquet/statistics-test.cc
+++ /home/travis/build/apache/arrow/cpp/src/parquet/statistics-test.cc (after clang format)
@@ -777,7 +777,8 @@
 TEST(TestStatisticsMinMax, Unsigned) {
   std::string dir_string(test::get_data_dir());
   std::stringstream ss;
-  ss << dir_string << "/" << "binary.parquet";
+  ss << dir_string << "/"
+     << "binary.parquet";
```

**Comment by zilder** [comment]
just did and pushed new version

**Review by majetideepak — APPROVED** [review]
+1 LGTM. I will merge after appveyor finishes. Thanks for this contribution!

**Comment by majetideepak** [comment]
@wesm I do not have the commit bit for Arrow. I will send an email to Jacques. Can you take care of merging this? Thanks.

**Pull request closed without merging**
